src/couch_httpd_auth.erl

```diff
@@ -495,3 +508,11 @@ integer_to_binary(Int, Len) when is_integer(Int), is_integer(Len) ->
     Padding = binary:copy(<<"0">>, Len),
     Padded = <<Padding/binary, Unpadded/binary>>,
     binary:part(Padded, byte_size(Padded), -Len).
+
+csrf_cookie(Req) ->
+    Secret = ?l2b(ensure_csrf_secret()),
+    <<Token:8/binary, _/binary>> = couch_uuids:random(),
+    Hmac = crypto:sha_mac(Secret, Token),
+    mochiweb_cookies:cookie("Csrf-token",
+        couch_util:encodeBase64Url(<<Token/binary,":",Hmac/binary>>),
+        [{path, "/"}] ++ max_age()).
```

Comment:
Some CSRF mitigation patterns do that but it's not the only method. We're doing this;

https://www.owasp.org/index.php/Cross-Site_Request_Forgery_%28CSRF%29_Prevention_Cheat_Sheet#Double_Submit_Cookies